modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Flamingos2.scala

```diff
@@ -91,24 +91,38 @@ object Flamingos2 {
   }
 
   private def disperserFromObserveType(observeType: String, d: Disperser): Grism = observeType match {
-    case DARK_OBSERVE_TYPE    => Grism.Dark
-    case _                    => d
+    case DARK_OBSERVE_TYPE => Grism.Dark
+    case _                 => d
   }
 
+  private def exposureTime(config: Config): ExtractFailure \/ Duration =
```

Comment:
My understanding is that you have `instrument:exposureTime`, `calibration:exposureTime`, and `observe:exposureTime`.  The `instrument` and `calibration` values won't necessarily agree but the "right" value to use will always be in `observe:exposureTime` which should match the one that corresponds to the step type.  In other words, I think this may be doing more work than necessary.